setup.py

```diff
@@ -844,6 +844,15 @@ def configure_extension_build():
             extra_compile_args += ['-g']
             extra_link_args += ['-g']
 
+    # special CUDA 11.7 package that requires installation of cuda runtime, cudnn and cublas
+    build_name = os.getenv('BUILD_NAME', '')
+    if package_type == 'manywheel' and 'cuda11_7-without-cudnn' in build_name:
+        report(f"Adding nvidia runtime for {build_name}")
+        extra_install_requires += [
+            'nvidia-cuda-runtime-cu11',
+            'nvidia-cudnn-cu11',
+            'nvidia-cublas-cu11']
+
```

Comment:
This feels fragile for many reasons:
 - It hardcodes specifics about binary builder process into setup.py (which supposed to be a generic script)
 - `nvidia-cudnn-cu11` dependencies should have a version constrains, otherwise its a time-bomb waiting for new and incompatible version of cublas to be published

In my opinion, a better approach would be to add logic to read and append requirements supplied via `PYTORCH_EXTRA_INSTALL_REQUIREMENTS` environment variable

Reply:
sounds good. 
PYTORCH_EXTRA_INSTALL_REQUIREMENTS - will set and passed from Github Action ?